### PR TITLE
Add civo provisioner

### PR DESCRIPTION
**Pull request by adamjohnson01**
- [x] I have raised an issue to propose this change.

## Description
Add Civo.com provisioner support, related to #32 

## How Has This Been Tested?

Operator created and deleted exit node on civo.com
```
2020/01/10 21:06:10 Inlets client: inlets/inlets:2.6.3
2020/01/10 21:06:10 Inlets pro: false
2020/01/10 21:06:11 Provisioning host with Civo
2020/01/10 21:06:11 Tunnel exists: nginx-1-tunnel
Instance ID: a6953d19-f25c-4ac4-8d42-e2558214e578
2020/01/10 21:06:14 Provisioning call took: 2.819603s
2020/01/10 21:06:14 Status (nginx-1): provisioning, ID: a6953d19-f25c-4ac4-8d42-e2558214e578, IP:
2020/01/10 21:06:41 Tunnel exists: nginx-1-tunnel
2020/01/10 21:07:11 Tunnel exists: nginx-1-tunnel
2020/01/10 21:07:11 Status (nginx-1): active, ID: a6953d19-f25c-4ac4-8d42-e2558214e578, IP: 91.211.152.226
2020/01/10 21:07:11 Tunnel exists: nginx-1-tunnel
2020/01/10 21:07:41 Tunnel exists: nginx-1-tunnel
2020/01/10 21:08:11 Tunnel exists: nginx-1-tunnel
2020/01/10 21:08:14 Delete event: nginx-1-tunnel default [{v1 Service nginx-1 28de566f-64ed-45ab-9ae0-28ef2a2bd0db 0x24a667b 0x24a667a}]
2020/01/10 21:08:14 Deleting exit-node for nginx-1: a6953d19-f25c-4ac4-8d42-e2558214e578, ip: 91.211.152.226
2020/01/10 21:08:20 Error updating service: nginx-1, services "nginx-1" not found
```

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/inlets/inlets/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

**Comment by alexellis** [comment]
Thank you for this PR

File: controller.go [hunk 1]
```diff
@@ -595,6 +617,28 @@ func (c *Controller) syncHandler(key string) error {
 				return err
 			}
 
+			if host.Status == provision.ActiveStatus {
```
Comment (alexellis) [inline controller.go]: Is this new?

Comment (adamjohnson01) [reply controller.go]: @alexellis , no it is in all of the others.

https://github.com/inlets/inlets-operator/blob/e39529b772a7d5da0b67cf39990141010d59e710/controller.go#L649

https://github.com/inlets/inlets-operator/blob/e39529b772a7d5da0b67cf39990141010d59e710/controller.go#L604

https://github.com/inlets/inlets-operator/blob/e39529b772a7d5da0b67cf39990141010d59e710/controller.go#L581

File: controller.go [hunk 2]
```diff
@@ -648,6 +670,28 @@ func (c *Controller) syncHandler(key string) error {
 			}
 			if host.Status == provision.ActiveStatus {
 
+				if host.IP != "" {
```
Comment (alexellis) [inline controller.go]: Is this needed specifically for Civo?

**Comment by alexellis** [comment]
@adamjohnson01 thanks for this, I've left a question for you and also need:

* The README to be updated to show the pricing, OS and etc used for Civo

Also, it looks like EC2 was merged, but that's also not listed in the README

**Comment by adamjohnson01** [comment]
@alexellis 

> @adamjohnson01 thanks for this, I've left a question for you and also need:
> 
> * The README to be updated to show the pricing, OS and etc used for Civo
> 
> Also, it looks like EC2 was merged, but that's also not listed in the README

@alexellis, I will update.

File: controller.go (hunk 2 above)
Comment (adamjohnson01) [reply controller.go]: No, this is present in all the other providers.

**Pull request merged**